### DISCUSSION TITLE
Read public assets hosted on Vercel

### DISCUSSION THREAD

**Discussion by jorisre** [remix-run/remix#3178] [Q&A]
Hi 👋🏻 
I'm using Remix with Vercel and I would like to resolve the public directory but I'm getting the following error: 

> 2022-05-13T08:05:41.933Z	0cea6eee-d5b6-4708-95be-c037ce4cac41	ERROR	Uncaught Exception 	{"errorType":"Error","errorMessage":"ENOENT: no such file or directory, open '/var/public/images/gift-card.png'","code":"ENOENT","errno":-2,"syscall":"open","path":"/var/public/images/gift-card.png","stack":["Error: ENOENT: no such file or directory, open '/var/public/images/gift-card.png'"]}
Unknown application error occurred

`/var/public` does not exist

The code: 
`path.resolve('./public', imgPath);`

Does anyone know how to solve this? Thanks !

**Comment by submetu** [comment]
I have the same issue

**Reply by JoeMethven** [reply]
me too :(

**Reply by jorisre** [reply]
I'm waiting for https://github.com/vercel/vercel/pull/8379

**Comment by sergiodxa** [comment]
If you check the deployment Output files the `public` folder is removed and assets inside are moved to the deployment root path.

So `path.resolve("./public", imgPath)` will always point to a path that doesn't exists anymore on Vercel.

<img width="727" alt="Screen Shot 2022-09-12 at 11 55 46" src="https://user-images.githubusercontent.com/1312018/189712395-6a3d9e16-281a-4036-812c-1d3b56d5f760.png">

**Reply by JoeMethven** [reply]
Yes, I saw this, but have tried numerous differing paths of "./locales/...", "/locales/...", "locales/..." (using remix-i18next), and none of them work, has anyone actually had a working path to a public resource in vercel?

**Reply by JoaoCnh** [reply]
I'm having this issue currently. The deployment function is running in a `/var/task/api` folder and I've inspected the contents of these folders and there's nothing there that is even similar to what is shown in the deployment Source tab which seems to indicate that it might not even be possible to access public files in a vercel deployment of Remix. I might have to change deployment platform
